sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -372,6 +425,31 @@ protected static SqlParts selectStatementForAndClause(List<Object> clause,
     }
 
     @SuppressWarnings("unchecked")
+    protected static SqlParts selectStatementForTextClause(Object clause,
+                                                           String indexName) {
+        if (clause == null) {
+            return null;  // no query here
+        }
+
+        if (indexName == null || indexName.isEmpty()) {
+            return null;
+        }
+
+        if (!(clause instanceof Map)) {
+            return null;  // should never get here as this would not pass normalization
+        }
+
+        Map<String, Object> textClause = (Map<String, Object>) clause;
+        Map<String, String> searchClause = (Map<String, String>) textClause.get(TEXT);
+
+        String tableName = IndexManager.tableNameForIndex(indexName);
+        String search = searchClause.get(SEARCH);
+
+        String sql = String.format("SELECT _id FROM %s WHERE %s MATCH ?", tableName, tableName);
+        return SqlParts.partsForSql(sql, new String[]{ String.format("'%s'", search) });
```

Comment:
Putting the single quotes in the sql as in `...MATCH '?'` causes an issue.  Escaping them in the sql String does not help.  The single quotes are necessary to tell SQLite to perform the FTS via the MATCH statement.  Instead I am now escaping all single quotes in the passed in `search` as two single quotes.  This resolves the SQL injection issue and allows for searching on text containing single quotes and apostrophes.  I added a test for this which will be included in the follow up PR.  Fixed in cdbf17d0a54e34272cbf77e02cdc4dc10c8b7921.